eth/pubsub.go

```diff
@@ -165,6 +184,349 @@ func (p *PubSub) handleBeaconBlock(ctx context.Context, msg *pubsub.Message) err
 	return nil
 }
 
+func (p *PubSub) handleAttestation(ctx context.Context, msg *pubsub.Message) error {
+	if msg == nil || msg.Topic == nil || *msg.Topic == "" {
+		return fmt.Errorf("nil message or topic")
+	}
+
+	attestation := ethtypes.Attestation{}
+	err := p.cfg.Encoder.DecodeGossip(msg.Data, &attestation)
+	if err != nil {
+		return fmt.Errorf("decode attestation gossip message: %w", err)
+	}
+
+	now := time.Now()
+	evt := &host.TraceEvent{
+		Type:      eventTypeHandleMessage,
+		PeerID:    p.host.ID(),
+		Timestamp: now,
+		Payload: map[string]any{
+			"PeerID":          msg.ReceivedFrom.String(),
+			"MsgID":           hex.EncodeToString([]byte(msg.ID)),
+			"MsgSize":         len(msg.Data),
+			"Topic":           msg.GetTopic(),
+			"Seq":             msg.GetSeqno(),
+			"CommIdx":         attestation.GetData().GetCommitteeIndex(),
+			"Slot":            attestation.GetData().GetSlot(),
+			"BeaconBlockRoot": attestation.GetData().GetBeaconBlockRoot(),
+			"Source":          attestation.GetData().GetSource(),
+			"Target":          attestation.GetData().GetTarget(),
+		},
+	}
+	slog.Info(
+		"Handling attestation gossip message",
+		"PeerID", p.host.ID(),
+		"RemotePeerID", msg.ReceivedFrom.String(),
+		"MsgID", hex.EncodeToString([]byte(msg.ID)),
+		"MsgSize", len(msg.Data),
+		"Topic", msg.GetTopic(),
+		"Seq", msg.GetSeqno(),
+		"CommIdx", attestation.GetData().GetCommitteeIndex(),
+		"Slot", attestation.GetData().GetSlot(),
+		"BeaconBlockRoot", attestation.GetData().GetBeaconBlockRoot(),
+		// There are other details in the Attestation message, add them when needed.
+	)
```

Comment:
I don't understand the benefits of logging this info using `slog` in addition to the above `evt`.

Reply:
I agree with @guillaumemichel , given the number of messages that we would receive on these topics (attestations, aggregations, sync_votes, etc), it would just spam the stdout

Reply:
Awesome thanks!